### ISSUE TITLE
'loss': 0.0 and 'grad_norm' : nan during the fine-tune

### ISSUE THREAD

**Issue by khanhdc** [JiuhaiChen/Florence-VL#9]
I used the scripts/florence-vl/llama/finetune_llama.sh to finetune the FlorenceVL, however, during the training process, the 'loss': 0.0 and 'grad_norm' : nan.
P/S: Since I have 1 GPU, I have converted the finetune_llama.sh to a .py file to run it. Please help with this. It would be greatly appreciated if you could provide instructions for the fintune process.

The converted script:
```
import os
import subprocess

# ===== Environment variables =====
os.environ["CUDA_DEVICE_MAX_CONNECTIONS"] = "1"

# ===== Paths =====
DATA_PATH = "/mnt/data/florence-2-vlm-sft-data/cambrian_sharegpt4v_vision_flan_docmatix.json"
IMG = "/mnt/data/florence-2-vlm-sft-data/"

CKPT_PATH = "/mnt/data/florence-2-vlm-pretrain/hub/models--jiuhai--florence-llama-pretrain/snapshots/b5c26b8c0048c394d8d8a5f91066702aa0bd9c07"
VIT_PATH = os.path.join(CKPT_PATH, "vision_tower")
OUTPUT = "/mnt/data/florence-2-vlm-pretrain/llava-llama-3-sft"

# ===== Hyperparameters =====
SAVE_PATH = "llama3-sft"
LEARNING_RATE = "1e-5"
NUM_EPOCHS = "1"

# ===== Build the training command =====
cmd = [
    "python", "llava/train/train_mem.py",
    "--model_name_or_path", CKPT_PATH,
    "--version", "llama3",
    "--data_path", DATA_PATH,
    "--image_folder", IMG,
    "--vision_tower", VIT_PATH,
    "--mm_projector_type", "mlp3x_gelu",
    "--mm_vision_select_layer", "-2",
    "--mm_use_im_start_end", "False",
    "--mm_use_im_patch_token", "False",
    "--image_aspect_ratio", "pad",
    "--group_by_modality_length", "True",
    "--bf16", "True",
    "--output_dir", OUTPUT,
    "--num_train_epochs", NUM_EPOCHS,
    "--per_device_train_batch_size", "8",
    "--per_device_eval_batch_size", "4",
    "--gradient_accumulation_steps", "2",
    "--eval_strategy", "no",
    "--save_strategy", "steps",
    "--save_steps", "30000",
    "--save_total_limit", "1",
    "--learning_rate", LEARNING_RATE,
    "--weight_decay", "0.",
    "--warmup_ratio", "0.03",
    "--lr_scheduler_type", "cosine",
    "--logging_steps", "1",
    "--tf32", "True",
    "--model_max_length", "2048",
    "--gradient_checkpointing", "True",
    "--dataloader_num_workers", "1",
    "--lazy_preprocess", "True",
    "--report_to", "wandb",
    "--run_name", SAVE_PATH,
]

# ===== Run training =====
print(">>> Starting single-GPU training (DeepSpeed disabled)...")
subprocess.run(cmd, check=True)

print("raining completed successfully.")
```
